### PR TITLE
Remove debug message when trying to open door

### LINKED ISSUE #13741: Debug message when trying to open door with makeshift picklock
When I tried to open metal door with my (.. makeshift picklock) and got my picklock broken, I get this:
default
 Win 10, tiles, 0.C-3774.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/27328214-debug-message-when-trying-to-open-door-with-makeshift-picklock?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Night-Pryanik**
Fixes #13741.

### CODE REVIEW

File: src/iuse_actor.cpp [hunk 1]
```diff
@@ -694,6 +694,7 @@ long pick_lock_actor::use( player *p, item *it, bool, const tripoint& ) const
         it->damage++;
         if( it->damage >= 5 ) {
             p->add_msg_if_player( m_bad, _( "The lock stumps your efforts to pick it, and you destroy your tool." ) );
+            p->i_rem(it);
```
Comment (Coolthulhu) [inline src/iuse_actor.cpp]: Removing it here means that the `it` pointer is invalidated.
This means that the weird check below (`it->damage < 100`? When will it be anything but true anyway?) invokes unspecified behavior, which could lead to crashes or at the very least random results.

File: src/iuse_actor.cpp [hunk 2]
```diff
@@ -707,11 +707,9 @@ long pick_lock_actor::use( player *p, item *it, bool, const tripoint& ) const
             g->add_event( EVENT_WANTED, int( calendar::turn ) + 300, 0, p->global_sm_location() );
         }
     }
-    // Special handling, normally the item isn't used up, but it is if broken.
     if( it->damage >= 5 ) {
-        return 1;
-    }
-
+        p->i_rem(it);
```
Comment (BevapDin) [inline src/iuse_actor.cpp]: And after the item is removed (and `it` is invalid), the function returns `it->type->charges_to_use()`, which accesses `it` again. There should probably be a `return 0` after `i_rem` or so.

**Comment by Coolthulhu** [comment]
Shouldn't it just return -1 instead of 1 or something like that?

Why would the lockpick use a totally different system from all the other items?

**Comment by Night-Pryanik** [comment]
Returning -1 just gives ?? item condition.

**Comment by BevapDin** [comment]
iuse functions return the number of charges to consume, 0 means no charges. No iuse function seems to return -1.

**Comment by illi-kun** [comment]
Please check how your changes affect power consumption of fingerpick CBM (I guess current PR can change its behavior introduced in pr #12243 but I'm not able to check it by myself since I'm out of my PC atm).

**Comment by Night-Pryanik** [comment]
My testing says it's all OK with fingerpicking. Correctly drains 25 power consumption per use.

**Comment by illi-kun** [comment]
...and drains nothing in case of cancellation / trying to open non-closed doors?

**Comment by Night-Pryanik** [comment]
Yeah, just so.

**Comment by illi-kun** [comment]
Great, thanks for checking it!

**Pull request merged**